tcms/core/management/commands/upgrade.py

```diff
@@ -0,0 +1,86 @@
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = "Perform post-upgrade tasks for Kiwi TCMS"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--noinput",
+            "--no-input",
+            "--yes",
+            "-y",
+            action="store_false",
+            dest="interactive",
+            help="Automatic mode. Does not require user confirmation",
+        )
+
+    def handle(self, *args, **kwargs):
+        output = None
+        if kwargs["verbosity"]:
+            output = self.stdout
+        answer = "y"
+        if kwargs["interactive"]:
+            answer = "x"
+
+        if output:
+            self.stdout.write(
+                """To finish the upgrade process, the following
+management commands will be executed:
+
+migrate
+refresh_permissions
+delete_stale_attachments
+delete_stale_comments
+remove_stale_contenttypes
+clean_orphan_obj_perms
+            """
+            )
+        if kwargs["interactive"]:
+            ok_to_proceed = input("Type 'yes' to continue, or 'no' to cancel: ")
+            if ok_to_proceed != "yes":
+                return ()
```

Comment:
This is wrong. Interactive mode should control whether or not individual sub-commands are asking for confirmation, e.g. to remove files or to update permissions, etc.  It doesn't make sense to allow the user to opt-out here. If they don't want to update they will simply not execute this command.